### PR TITLE
Rewrite running multiple tests for browsers and devices

### PR DESCRIPTION
**Pull request by mmarkelov**

### CODE REVIEW

**Review by mxschmitt — APPROVED** [review]
cool, lets create a rc and test it on our example repository?

**Comment by mmarkelov** [comment]
@mxschmitt sure. But it's not ready yet. I just working on it. Just now thinking about to add some option to control it, cause I don't want to break current behaviour. With this option you will be able to run tests for devices using it `defaultBrowserType`

File: src/PlaywrightRunner.ts [hunk 1]
```diff
@@ -81,67 +103,95 @@ class PlaywrightRunner extends JestRunner {
...
-            ),
+          }
+        }
+      } else {
+        for (const browser of browsers) {
+          checkBrowserEnv(browser)
+          const { devices: availableDevices, instance } = getPlaywrightInstance(
+            browser,
+          )
+          const resultDevices = getDevices(devices, availableDevices)
```
Comment (ValeryVS) [inline src/PlaywrightRunner.ts:165]: It seems, that there will be error if custom devices are used.

With such config
```
  browsers: ['chromium', 'firefox', 'webkit'],
  devices: [
    'iPad Pro 11 landscape',
    'iPhone SE',
    {
      name: 'FHD desktop',
      viewport: {
        width: 1920,
        height: 1080,
      },
      deviceScaleFactor: 1,
      isMobile: false,
      hasTouch: false,
      defaultBrowserType: 'webkit',
    },
    {
      name: 'FHD desktop',
      viewport: {
        width: 1920,
        height: 1080,
      },
      deviceScaleFactor: 1,
      isMobile: false,
      hasTouch: false,
      defaultBrowserType: 'firefox',
    },
  ],
```

Test will be runned on webkit-ipad, webkit-iphone, and two times on chromium desktop.

Comment (ValeryVS) [reply src/PlaywrightRunner.ts:165]: #464 464

**Pull request merged**